### ISSUE TITLE
3.0 Update Make HTTP requests using IHttpClientFactory, use System.Text.Json serializers

### ISSUE THREAD

**Issue by HBTeun** [aspnet/AspNetCore.Docs#15273]
Please show samples using System.Text.Json serializer.

Many examples use the ReadAsAsync&lt;T&gt;() extension, but this extension is not available without adding Microsoft.AspNet.WebApi.Client Nuget package. This creates confusion. It also uses Newtonsoft.Json internally.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 881cef77-11ef-ac11-06d0-0996a1ccfa1b
* Version Independent ID: 3d356334-e672-7e8b-1fef-7944ebee99eb
* Content: [Make HTTP requests using IHttpClientFactory in ASP.NET Core](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/http-requests?view=aspnetcore-3.0#feedback)
* Content Source: [aspnetcore/fundamentals/http-requests.md](https://github.com/aspnet/AspNetCore.Docs/blob/master/aspnetcore/fundamentals/http-requests.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-fundamentals**
* GitHub Login: @stevejgordon
* Microsoft Alias: **scaddie**

**Comment by Rick-Anderson** [comment]
![image](https://user-images.githubusercontent.com/3605364/67430748-9ff79c80-f57e-11e9-9da4-1dfac63648f8.png)

@serpent5  this looks like a fun one if you're interested.

**Comment by HBTeun** [comment]
I'm sorry, I had not looked in the closed issue's and missed the other tickets. But there should be a note about the nuget reference.
Still I think it's a good idee to have at least an example using System.Text.Json serialization.

**Comment by Rick-Anderson** [comment]
@HBTeun  is this the comment?

You can make use of the new json serializer from microsoft. The code using `ReadAsAsync` would then be refactored as:

``` 
using (var contentStream = await response.Content.ReadAsStreamAsync())
{
    PullRequests = await JsonSerializer.DeserializeAsync<IEnumerable<GitHubPullRequest>>(contentStream);
}
```
https://docs.microsoft.com/en-us/dotnet/api/system.text.json?view=netcore-3.0


We should add that.

**Comment by serpent5** [comment]
Sure, I can do this. That example above is how I've seen this done too. Should we version for 3.0 and switch to this approach or just add an example of this in its own section?

**Comment by Rick-Anderson** [comment]
If possible, do this

```
::: moniker range=">= aspnetcore-3.0"
// Copy exiting MDF file here, and update to 3.0
// Add new code
::: moniker-end

::: moniker range="< aspnetcore-3.0"
// Original file here
::: moniker-end
```

**Comment by HBTeun** [comment]
@Rick-Anderson Yes, that was the comment I meant.

**Comment by serpent5** [comment]
@Rick-Anderson Do nested monikers work? There are a few monikers in this file. `<= aspnetcore-2.2` and `>= aspnetcore-2.2` in the content and `>= aspnetcore-2.1` in the metadata. Moving the original content under a `< aspnetcore-3.0` will introduce nesting and I just want to check whether that's supported.

**Comment by Rick-Anderson** [comment]
No, nested monikers are not supported. That's why we've been using the following pattern:

```
::: moniker range=">= aspnetcore-3.0"
// Copy exiting MDF file here, and update to 3.0
// Add new code
::: moniker-end

::: moniker range="< aspnetcore-3.0"
// Original file here
::: moniker-end
```

**Comment by Rick-Anderson** [comment]
> `>= aspnetcore-2.1` in the metadata. 

That should be removed. 

For the 3.0 version, just show `System.Text.Json`

**Comment by Rick-Anderson** [comment]
It looks like this one requires

```
::: moniker range=">= aspnetcore-3.0"
// Copy exiting MDF file here, and update to 3.0
// Add new code
::: moniker-end

::: moniker range="= aspnetcore-2.2"
//  copy of doc with 2.1 removed
::: moniker-end

::: moniker range="<= aspnetcore-2.2"
//  copy of doc with 2.2 removed
::: moniker-end
```

cc @scottaddie

**Comment by Rick-Anderson** [comment]
@serpent5  I can split this up for you into 3 sections.

**Comment by Rick-Anderson** [comment]
@serpent5 I merged #15329 so you can start anytime.

**Comment by serpent5** [comment]
@Rick-Anderson Brilliant. Thanks for doing this.
